lib/stores/S3Store.ts

```diff
@@ -226,6 +244,49 @@ export default class S3Store extends DataStore {
     return data.ETag as string
   }
 
+  private async uploadIncompletePart(
+    id: string,
+    readStream: fs.ReadStream | Readable
+  ): Promise<string> {
+    const data = await this.client
+      .putObject({
+        Bucket: this.bucket,
+        Key: id,
+        Body: readStream,
+      })
+      .promise()
+    return data.ETag as string
+  }
+
+  private async getIncompletePart(id: string): Promise<Buffer | undefined> {
+    try {
+      const data = await this.client
+        .getObject({
+          Bucket: this.bucket,
+          Key: id,
+        })
+        .promise()
+      return data.Body as Buffer
+    } catch {
+      return undefined
```

Comment:
We could have more fine-graned error handling here. Only if the object was not found, undefined would be returned. Otherwise the error is raised